encord/objects/classification_instance.py

```diff
@@ -95,9 +107,67 @@ def _last_frame(self) -> Union[int, float]:
         else:
             return self._parent.number_of_frames
 
+    @property
+    def range_list(self) -> Ranges:
+        if self._range_only or (self._parent is not None and self._parent.data_type == DataType.AUDIO):
+            return self._range_manager.get_ranges()
+        else:
+            raise LabelRowError("No ranges available for this classification instance.")
+
+    def is_range_only(self) -> bool:
+        return self._range_only
+
     def is_assigned_to_label_row(self) -> bool:
         return self._parent is not None
 
+    def _set_for_ranges(
+        self,
+        frames: Frames,
+        overwrite: bool,
+        created_at: Optional[datetime],
+        created_by: Optional[str],
+        confidence: float,
+        manual_annotation: bool,
+        last_edited_at: Optional[datetime],
+        last_edited_by: Optional[str],
+        reviews: Optional[List[dict]],
+    ):
+        new_range_manager = RangeManager(frame_class=frames)
+        conflicting_ranges = self._is_classification_already_present_on_range(new_range_manager.get_ranges())
+        if conflicting_ranges and not overwrite:
+            raise LabelRowError(
+                f"The classification '{self.classification_hash}' already exists "
+                f"on the ranges {conflicting_ranges}. "
+                f"Set 'overwrite' parameter to True to override."
+            )
+
+        ranges_to_add = new_range_manager.get_ranges()
+        for range_to_add in ranges_to_add:
+            self._check_within_range(range_to_add.end)
+
+        """
+        At this point, this classification instance operates on ranges, NOT on frames.
+        We therefore leave only FRAME 0 in the map.The frame_data for FRAME 0 will be
+        treated as the data for all "frames" in this classification instance.
+        """
+        self._set_frame_and_frame_data(
+            frame=0,
+            overwrite=True,  # We always overwrite the frame data here because it represents the entire instance
+            created_at=created_at,
+            created_by=created_by,
+            confidence=confidence,
+            manual_annotation=manual_annotation,
+            last_edited_at=last_edited_at,
+            last_edited_by=last_edited_by,
+            reviews=reviews,
+        )
```

Comment:
I am very worried about this, but can't think of a good way around it.

If I am working on a `range_only=True` classification instance, I ensure the `frames_to_data_map` only has ONE frame, and that frame will represent the entire classification instance.

Given that I am always operating on "Frame 0", I will have to overwrite this each time someone updates the classification instance. This is problematic because somehow might not know they are overwriting the FrameData for the ENTIRE classification instance.

The only way I can see myself getting around this, is to add a `ranges_to_data_map`, and have the data returned be for that particular range.

Reply:
Should we just say that "frame parameter is ignored for Audio files" and throw an error if anything except 0 is passed when data type is Audio? That way it would be clear that frames are not applicable here, and this "zeroth frame" approach would be just a detail of the implementation?

Reply:
Done!